src/Camera/QGCCameraManager.cc

```diff
@@ -555,3 +568,103 @@ QGCCameraManager::_stepStream(int direction)
         }
     }
 }
+
+const QVariantList &QGCCameraManager::cameraList()
+{
+    if (_cameraList.isEmpty()) {
+        const QList<CameraMetaData*> cams = _parseCameraMetaData(QStringLiteral(":/json/CameraMetaData.json"));
+        _cameraList.reserve(cams.size());
+
+        for (CameraMetaData* cam : cams) {
+            _cameraList << QVariant::fromValue(cam);
+        }
+    }
+
+    return _cameraList;
+}
+
+QList<CameraMetaData*> QGCCameraManager::_parseCameraMetaData(const QString &jsonFilePath)
+{
+    QList<CameraMetaData*> cameraList;
+
+    QFile file(jsonFilePath);
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) {
```

Comment:
You need to use JsonHelper::openInternalQGCJsonFile to get the translation support. This will do this for you:
* Open file
* Validation
* Transation

Reply:
I'll get it right eventually